oak_restricted_kernel/src/dice_attestation.rs

```diff
@@ -15,43 +15,38 @@
 //
 
 use coset::CborSerializable;
+use oak_dice::evidence::ApplicationKeys;
 use zerocopy::FromZeroes;
 
-/// Generates an ECA certificate for use by the application.
-fn generate_application_certificate(
-    kernel_signing_key: &p256::ecdsa::SigningKey,
-    kernel_cert_issuer: alloc::string::String,
-    application_verifying_key: &p256::ecdsa::VerifyingKey,
-    app_digest: &[u8],
-) -> coset::CoseSign1 {
-    let additional_claims = alloc::vec![(
-        coset::cwt::ClaimName::PrivateUse(oak_dice::cert::LAYER_2_CODE_MEASUREMENT_ID),
-        coset::cbor::value::Value::Bytes(app_digest.into()),
-    )];
-    oak_dice::cert::generate_signing_certificate(
-        kernel_signing_key,
-        kernel_cert_issuer,
-        application_verifying_key,
-        additional_claims,
-    )
-    .expect("couldn't generate signing certificate")
+fn certificate_to_owned_slice(
+    cert: coset::sign::CoseSign1,
+) -> [u8; oak_dice::evidence::CERTIFICATE_SIZE] {
+    let vec = cert.to_vec().expect("couldn't serialize certificate");
+    let slice = [u8; oak_dice::evidence::CERTIFICATE_SIZE];
+    slice[..vec.len()].copy_from_slice(&vec);
+    slice
+}
+
+fn private_key_to_owned_slice(
+    key: SigningKey<NistP256>,
```

Comment:
this isn't a new concern, but afaik when the relevant key struct is dropped, the key is not zeroed out in memory. We do this for our own key structs, but the libraries do not.

Reply:
The SigningKey will actually also zero its content on drop. I don't think this function is needed since it should be possible to create an array by just using `key.to_bytes().into()`

Reply:
ah yes, confirmed by looking at the ecdsa crate source